tests/base.go

```diff
@@ -2,65 +2,56 @@ package tests
 
 import (
 	"bytes"
-	"fmt"
-	"io/ioutil"
-	"log"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
 	"net"
 	"os"
 	"os/exec"
 	"strings"
 	"time"
 
-	"github.com/goombaio/namegenerator"
+	"github.com/pkg/errors"
 	"golang.org/x/crypto/ssh"
 )
 
-// UpWg used for up wireguard
-func UpWg(wgConfig string) {
+// UpWg used for setting up the wireguard interface
+func UpWg(wgConfig string) (string, error) {
 	f, err := os.Create("/tmp/test.conf")
-
 	if err != nil {
-		log.Fatal(err)
+		return "", errors.Wrapf(err, "error creating file")
 	}
-
 	defer f.Close()
 
-	_, err2 := f.WriteString(wgConfig)
-
-	if err2 != nil {
-		log.Fatal(err2)
+	_, err = f.WriteString(wgConfig)
+	if err != nil {
+		return "", errors.Wrapf(err, "error writing wireguard config string")
 	}
 
 	cmd := exec.Command("sudo", "wg-quick", "up", "/tmp/test.conf")
 	stdout, err := cmd.Output()
-
 	if err != nil {
-		fmt.Println(err)
-		return
+		return "", errors.Wrapf(err, "error excuting wg-quick up")
 	}
-
-	// Print the output
-	fmt.Println(string(stdout))
+	return string(stdout), nil
 }
 
-// DownWG used for down wireguard
-func DownWG() {
+// DownWG used for tearing down the wireguard interface
+func DownWG() (string, error) {
 	cmd := exec.Command("sudo", "wg-quick", "down", "/tmp/test.conf")
 	stdout, err := cmd.Output()
-
 	if err != nil {
-		fmt.Println(err)
-		return
+		return "", errors.Wrapf(err, "error excuting wg-quick down ")
 	}
-	fmt.Println(string(stdout))
+	return string(stdout), nil
 }
 
-// RemoteRun used for ssh host
-func RemoteRun(user string, addr string, cmd string) (string, error) {
-	privateKey := os.Getenv("PRIVATEKEY")
-	key, err := ssh.ParsePrivateKey([]byte(privateKey))
+// RemoteRun used for running cmd remotly using ssh
+func RemoteRun(user string, addr string, cmd string, sk string) (string, error) {
```

Comment:
should make the function 
`func RemoteRun(user, addr, cmd, privateKey string)`
also privateKey should be explicit as privateKey not sk, so it won't get confused with `secretKey`